src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1598,7 +1596,68 @@ SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
 		case Anum_pg_dist_node_metadatasynced:
 		{
 			ErrorIfCoordinatorMetadataSetFalse(workerNode, value, "metadatasynced");
+			metadataSyncCommand = NodeMetadataSyncedUpdateCommand(workerNode->nodeId,
+																  DatumGetBool(value));
+			break;
+		}
+
+		default:
+		{
+			ereport(ERROR, (errmsg("could not find valid entry for node \"%s:%d\"",
+								   workerNode->workerName, workerNode->workerPort)));
+		}
+	}
+
+	SendCommandToWorkersWithMetadata(metadataSyncCommand);
+
+	return workerNode;
+}
+
+
+/*
+ * SetWorkerColumnOptional function sets the column with the specified index
+ * on the worker in pg_dist_node, by calling SetWorkerColumnLocalOnly.
+ * It also sends the same command optionally for node update to other metadata nodes.
+ * Returns the new worker node after the modification.
+ */
+WorkerNode *
+SetWorkerColumnOptional(WorkerNode *workerNode, int columnIndex, Datum value)
+{
+	workerNode = SetWorkerColumnLocalOnly(workerNode, columnIndex, value);
```

Comment:
I don't think we should have the duplicate function for `SetWorkerColumnOptional`. It should share as much as possible with `SetWorkerColumn` and only the command sending phase should be different.

Can you work on merging these two code-paths to a common function? I think having  a static function `SetWorkerColumnInternal( ... , bool raiseErrors) ` would be acceptable.

I know that I pushed you to avoid having `boolean` flag, but I mostly try to avoid that in external APIs. The internal/static functions are more acceptable to have boolean functions